packages/shared/src/components/checklist/ChecklistCard.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ReactElement, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { Card } from '../cards/Card';
+import MiniCloseIcon from '../icons/MiniClose';
+import { IconSize } from '../Icon';
+import { ChecklistStep } from './ChecklistStep';
+import { ChecklistAction, ChecklistCardProps } from '../../lib/checklist';
+
+const ChecklistCard = ({
+  className,
+  title,
+  description,
+  steps,
+  onRequestClose,
+}: ChecklistCardProps): ReactElement => {
+  const activeStep = useMemo(
+    () => steps.find((item) => !item.action.dateCompleted)?.action.type,
+    [steps],
+  );
+  const [checkedStep, setCheckedStep] = useState<string>(activeStep);
+  const isDone = useMemo(
+    () => steps.every((item) => !!item.action.dateCompleted),
+    [steps],
+  );
```

Comment:
This almost sounds like a perfect fit for a hook, especially since a lot of this will be controlled by external placed actions.
What do you think?

Reply:
Yes, as written in description, some of the logic should go to useActions hook, its here until we see how that one turns out.